pkg/network/ebpf/c/protocols/tls/https.h

```diff
@@ -133,10 +154,16 @@ static __always_inline void init_ssl_sock(void *ssl_ctx, u32 socket_fd) {
 
 static __always_inline void map_ssl_ctx_to_sock(struct sock *skp) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
-    void **ssl_ctx_map_val = bpf_map_lookup_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
-    if (ssl_ctx_map_val == NULL) {
+    ssl_handshake_args_t *args = bpf_map_lookup_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
+    if (args == NULL) {
         return;
     }
+    if (args->timestamp > 0) {
+        if ((bpf_ktime_get_ns() - args->timestamp) > ssl_async_handshake_window()) { /* 500 us by default */
+            return;
+        }
+    }
```

Comment:
what are you doing here?

Reply:
I drop the association if it's out of window (timeout) as we can't be sure 100% that the right socket, this mechanism is to avoid miss association